lib/duplicate_deleter.rb

```diff
@@ -62,8 +62,13 @@ def call(ids, id_type: 'content_id')
 
       content_ids = results[:results].map { |a| a['content_id'] }
       if content_ids.uniq.count != 1
-        io.puts "Skipping #{id_type} #{id} as multiple content_id's detected #{content_ids.uniq.join(', ')}"
-        next
+        if results[:results].any? { |a| a[:elasticsearch_type] != type_to_delete && a['content_id'].nil? }
+          io.puts "Skipping #{id_type} #{id} as it is indexed with a valid '_type' but a missing content ID"
+          next
+        elsif results[:results].any? { |a| a[:elasticsearch_type] == type_to_delete && !a['content_id'].nil? }
```

Comment:
I think this ok. If anything it will be too strict and still skip too much, but if so we can raise another PR.

These are the different combinations I'm thinking of

1. Deleted type has non-nils, Non-deleted type(s) has all nils = bad, we've lost information
2. Deleted type has non-nils, non-deleted type(s) has all non-nils = ok
3. Deleted type has non-nils, non-deleted types has a mix = unknown
4. Deleted type has all nils = should always be ok

😖